packages/backend/src/services/RenameService/RenameService.ts

```diff
@@ -0,0 +1,695 @@
+import {
+    AlreadyExistsError,
+    ApiRenameBody,
+    ApiRenameChartBody,
+    ApiRenameResponse,
+    ChartSummary,
+    DashboardDAO,
+    Explore,
+    ForbiddenError,
+    getDeepestPaths,
+    getErrorMessage,
+    getFieldRef,
+    getItemId,
+    isDashboardChartTileType,
+    isExploreError,
+    isSubPath,
+    NameChanges,
+    NotFoundError,
+    ParameterError,
+    PromotedChart as PromotedChangeChart,
+    PromotedSpace,
+    PromotionAction,
+    PromotionChanges,
+    RenameResourcesPayload,
+    RenameType,
+    RequestMethod,
+    SavedChartDAO,
+    SCHEDULER_TASKS,
+    SchedulerAndTargets,
+    SessionUser,
+    Space,
+    SpaceShare,
+    SpaceSummary,
+    UnexpectedServerError,
+} from '@lightdash/common';
+import { cloneDeep } from 'lodash';
+
+import { LightdashAnalytics } from '../../analytics/LightdashAnalytics';
+import { LightdashConfig } from '../../config/parseConfig';
+import Logger from '../../logging/logger';
+import { DashboardModel } from '../../models/DashboardModel/DashboardModel';
+import { ProjectModel } from '../../models/ProjectModel/ProjectModel';
+import { SavedChartModel } from '../../models/SavedChartModel';
+import { SchedulerModel } from '../../models/SchedulerModel';
+import { SpaceModel } from '../../models/SpaceModel';
+import { SchedulerClient } from '../../scheduler/SchedulerClient';
+import { BaseService } from '../BaseService';
+import {
+    getNameChanges,
+    renameAlert,
+    renameDashboard,
+    renameDashboardScheduler,
+    renameSavedChart,
+} from './rename';
+
+type RenameServiceArguments = {
+    lightdashConfig: LightdashConfig;
+    analytics: LightdashAnalytics;
+    projectModel: ProjectModel;
+    savedChartModel: SavedChartModel;
+    dashboardModel: DashboardModel;
+    schedulerClient: SchedulerClient;
+    schedulerModel: SchedulerModel;
+};
+
+export class RenameService extends BaseService {
+    private readonly lightdashConfig: LightdashConfig;
+
+    private readonly savedChartModel: SavedChartModel;
+
+    private readonly analytics: LightdashAnalytics;
+
+    private readonly projectModel: ProjectModel;
+
+    private readonly dashboardModel: DashboardModel;
+
+    private readonly schedulerClient: SchedulerClient;
+
+    private readonly schedulerModel: SchedulerModel;
+
+    constructor(args: RenameServiceArguments) {
+        super();
+        this.lightdashConfig = args.lightdashConfig;
+        this.analytics = args.analytics;
+        this.savedChartModel = args.savedChartModel;
+        this.projectModel = args.projectModel;
+        this.dashboardModel = args.dashboardModel;
+        this.schedulerClient = args.schedulerClient;
+        this.schedulerModel = args.schedulerModel;
+    }
+
+    public async getFieldsForChart({
+        user,
+        projectUuid,
+        chartUuid,
+    }: {
+        user: SessionUser;
+        projectUuid: string;
+        chartUuid: string;
+    }) {
+        const chart = await this.savedChartModel.get(chartUuid);
```

Comment:
these are calling model methods directly without a permission check

Reply:
I commented here since this is a public method. maybe switch to private if it not intended to use in a controller